packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -189,9 +201,219 @@ const renderQueryEditorHeaderLabel = (
   return null;
 };
 
+const QueryLoader = observer(
+  (props: {
+    editorStore: QueryEditorStore;
+    applicationStore: ApplicationStore<
+      LegendQueryApplicationConfig,
+      LegendQueryApplicationPlugin
+    >;
+  }) => {
+    const { editorStore, applicationStore } = props;
+    const queryFinderRef = useRef<SelectComponent>(null);
+    const [selectedQueryID, setSelectedQueryID] = useState('');
+    const queryLoaderColumns = ['Name', 'Author', 'Version'];
+    const closeQueryImporter = (): void => {
+      editorStore.setIsQueryLoaderOpen(false);
+    };
+    const handleEnterQueryImporter = (): void =>
+      queryFinderRef.current?.focus();
+    const loadSelectedQuery = (): void => {
+      if (selectedQueryID) {
+        editorStore.setIsQueryLoaderOpen(false);
+        applicationStore.navigator.goTo(
+          generateExistingQueryEditorRoute(selectedQueryID),
+        );
+        window.location.reload();
+      } else {
+        applicationStore.notifyWarning(
+          'No query is selected. Please select a query to load or close.',
+        );
+      }
+    };
+    const buildQueryTableRow = (query: LightQuery): string[] => [
+      query.name,
+      query.owner ?? 'anonymous',
+      query.versionId,
+    ];
+    const searchInputRef = useRef<HTMLInputElement>(null);
+    const loadQueries = (searchText: string): void => {
+      flowResult(editorStore.loadQueries(searchText))
+        .then(() => editorStore.setIsLoadingQueryInProgress(false))
+        .catch(applicationStore.alertUnhandledError);
+    };
+    const searchQuery: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+      editorStore.setSearchText(event.target.value);
+      loadQueries(editorStore.searchText);
+    };
+    const clearQuerySearching = (): void => {
+      editorStore.setSearchText('');
+      loadQueries(editorStore.searchText);
+    };
+    const toggleIsMineOnly = (): void => {
+      editorStore.setIsMineOnly(!editorStore.isMineOnly);
+    };
+
+    return (
+      <Dialog
+        open={editorStore.isQueryLoaderOpen}
+        onClose={closeQueryImporter}
+        TransitionProps={{
+          onEnter: handleEnterQueryImporter,
+        }}
+        classes={{ container: 'search-modal__container' }}
+        PaperProps={{ classes: { root: 'search-modal__inner-container' } }}
+      >
+        <div className="modal modal--dark search-modal">
+          <div className="modal__title">Select a Query to Load</div>
+          <div className="query-editor__query-loader__filter-section">
+            <div className="query-editor__query-loader__filter-section__header__label">
```

Comment:
Hm, I would say for now, let's remove this, it looks similar to the header of the modal. Please remove the CSS class accordingly